code/modules/virus2/helpers.dm

```diff
@@ -112,6 +112,8 @@ proc/airborne_can_reach(turf/source, turf/target, var/radius=5)
 	return FALSE
 
 /proc/attempt_colony(var/atom/A, var/datum/disease2/disease/D,var/info)
+	if(!D || !A)
```

Comment:
what is calling this with either argument null?

Reply:
/obj/item/weapon/virusdish/attack_self was calling it. i fixed it, but i am scared someone will fuck it up again.

Reply:
via /obj/item/weapon/virusdish/infection_attempt, that is.

Reply:
Isn't a runtime in here the only thing that led you to discover this bug? If you add these checks you will not know if someone fucked up again, if I'm understanding correctly.

Reply:
If someone in the future commits code that isn't allowed to function due to a null check, it's just proof they didn't test it.
I'll get rid of it though